### PR TITLE
newFeature: Add configuration option to control case-insensitive under oracle tenant

### LINKED ISSUE #64: [Feature]: Add parameters to control handling of lowercase when SQL concatenation
### Describe your use case

In Oceanbase, under the Oracle tenant, [uppercase is created in lowercase](https://www.oceanbase.com/docs/common-oceanbase-database-cn-1000000000643775). Some SQL editors will automatically perform syntax conversion when querying, but in FlinkSQL, if table names and column names are all lowercase in DDL, while writing data in FlinkSQL must be uppercase. Unlike the habit of definition at the same time, writing in lowercase will fail.

### Describe the solution you'd like

We can add parameters in oceanbase connector to automate it.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by yuanoOo**
## Summary

close #64 

## Solution Description

- This issue only exists under oracle tenant
- There are no integration tests in the code since OceanBase enterprise version doesn't seem to provide a docker image.

### CODE REVIEW

File: flink-connector-oceanbase/src/test/java/com/oceanbase/connector/flink/dialect/MySQLDialectUnitTest.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+ */
+package com.oceanbase.connector.flink.dialect;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+public class MySQLDialectUnitTest {
+
+    @Test
+    public void testQuoteIdentifier() {
```
Comment (whhe) [inline flink-connector-oceanbase/src/test/java/com/oceanbase/connector/flink/dialect/MySQLDialectUnitTest.java:24]: You should create a dialect instance and test its method returns.

File: flink-connector-oceanbase/src/main/java/com/oceanbase/connector/flink/OceanBaseConnectorOptions.java [hunk 2]
```diff
@@ -118,6 +118,13 @@ public class OceanBaseConnectorOptions extends ConnectorOptions {
                     .defaultValue(Duration.ofSeconds(30))
                     .withDescription("Client heartbeat timeout in direct load task.");
 
+    public static final ConfigOption<Boolean> ORACLE_TENANT_CASE_INSENSITIVE =
+            ConfigOptions.key("oracle.tenant.case.insensitive")
```
Comment (whhe) [inline flink-connector-oceanbase/src/main/java/com/oceanbase/connector/flink/OceanBaseConnectorOptions.java:122]: It's better to use a prefix with `table`, like `table.case-insensitive`, as there may be other table options in the future.

**Comment by whhe** [comment]
Thanks for your contribution! I left some comments.

Before merging this pr, you also need to sign the CLA following the CLAassistant's comment above. And it's also necessary to add an integrated test case for the new introcuded option.

**Comment by yuanoOo** [comment]
> Thanks for your contribution! I left some comments.
> 
> Before merging this pr, you also need to sign the CLA following the CLAassistant's comment above. And it's also necessary to add an integrated test case for the new introcuded option.

- 1、I have already signed the CLA, but the page keeps showing that I have not signed it...
- 2、About integration testing, the oceanbase community version does not support Oracle tenants, and the enterprise version does not seem to provide docker images. Do you have any advice。

**Comment by whhe** [comment]
> > Thanks for your contribution! I left some comments.
> > Before merging this pr, you also need to sign the CLA following the CLAassistant's comment above. And it's also necessary to add an integrated test case for the new introcuded option.
> 
> * 1、I have already signed the CLA, but the page keeps showing that I have not signed it...
> * 2、About integration testing, the oceanbase community version does not support Oracle tenants, and the enterprise version does not seem to provide docker images. Do you have any advice。

The CLA check is passed now. For testing part, you can add a test case, execute it locally but skip it in GitHub CI, just like [OceanBaseOracleDynamicTableSinkITCase](https://github.com/apache/flink-connector-jdbc/blob/main/flink-connector-jdbc/src/test/java/org/apache/flink/connector/jdbc/databases/oceanbase/table/OceanBaseOracleDynamicTableSinkITCase.java).

File: flink-connector-oceanbase/src/test/java/com/oceanbase/connector/flink/dialect/MySQLDialectUnitTest.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+ */
+package com.oceanbase.connector.flink.dialect;
+
+import com.oceanbase.connector.flink.OceanBaseConnectorOptions;
+
+import org.junit.Assert;
+import org.junit.Test;
+import org.testcontainers.shaded.com.google.common.collect.Maps;
+
+public class MySQLDialectUnitTest {
```
Comment (whhe) [inline flink-connector-oceanbase/src/test/java/com/oceanbase/connector/flink/dialect/MySQLDialectUnitTest.java:24]: Should be `OceanBaseOracleDialectTest`.

File: flink-connector-oceanbase/src/test/java/com/oceanbase/connector/flink/OceanBaseOracleConnectorITCase.java [hunk 4]
```diff
@@ -0,0 +1,537 @@
...
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+@Ignore
+public class OceanBaseOracleConnectorITCase extends OceanBaseOracleTestBase {
```
Comment (whhe) [inline flink-connector-oceanbase/src/test/java/com/oceanbase/connector/flink/OceanBaseOracleConnectorITCase.java:69]: There should be a test case for case-insensitive option, and I think other cases are not necessary for now.

File: flink-connector-oceanbase-base/src/test/java/com/oceanbase/connector/flink/OceanBaseOracleTestBase.java [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+public abstract class OceanBaseOracleTestBase extends TestLogger {
```
Comment (whhe) [inline flink-connector-oceanbase-base/src/test/java/com/oceanbase/connector/flink/OceanBaseOracleTestBase.java:30]: Maybe we should rename the existing test classes to "OceanBaseMySQLxxxxxx".

File: flink-connector-oceanbase/src/test/java/com/oceanbase/connector/flink/dialect/MySQLDialectUnitTest.java (hunk 3 above)
Comment (yuanoOo) [reply flink-connector-oceanbase/src/test/java/com/oceanbase/connector/flink/dialect/MySQLDialectUnitTest.java]: get it

File: flink-connector-oceanbase/src/test/java/com/oceanbase/connector/flink/OceanBaseOracleConnectorITCase.java (hunk 4 above)
Comment (yuanoOo) [reply flink-connector-oceanbase/src/test/java/com/oceanbase/connector/flink/OceanBaseOracleConnectorITCase.java:61]: I kept two basic test cases and they passed in the dev env。

**Review by whhe — APPROVED** [review]

**Pull request merged**